public/app/features/datasources/DataSourcesListPage.tsx

```diff
@@ -58,6 +66,12 @@ export class DataSourcesListPage extends PureComponent<Props> {
     return await this.props.loadDataSources();
   }
 
+  deleteDataSource = async (id: number) => {
+    await this.props.loadDataSource(id);
+    await this.props.deleteDataSource();
```

Comment:
This looks strange, can’t we have deleteDashboard take an id as parameter instead of having to load it first ? 

Does this have a confirm step? 

Want to do a quick test and review of this early next week so please do not merge before then

Reply:
This seems to reuse the `deleteDataSource` action from the (single) datasource settings page, where the redux state has loaded a datasource. I agree that this looks non-intuitive, and I'd also preferred it if the loading in line 70 was not needed for the sole purpose of deleteing a datasource. Can we change this to make `deleteDataSource` not dependent on redux state?

Reply:
Yep, it was using the same logic from datasource settings page. Already rewrote it.